python/cudf/cudf/core/column/column.py

```diff
@@ -1403,10 +1403,11 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
                 )
             except (pa.ArrowInvalid, pa.ArrowTypeError, TypeError):
                 if is_categorical_dtype(dtype):
-                    data = as_column(
-                        pd.Series(arbitrary, dtype="category"),
-                        nan_as_null=nan_as_null,
-                    )
+                    sr = pd.Series(arbitrary, dtype="category")
+                    data = as_column(sr, nan_as_null=nan_as_null)
+                elif np_type == np.str_:
```

Comment:
Do we need to handle `object` dtypes here as well or just `str`?

Reply:
Right now, both string and object cases will fail all the ifs and we fall back on `np.array()` with a `dtype` argument to prep the data for `as_column`. In the object case, everything ends up being cast to `int64` typed numba array under the hood with a zero (or potentially arbitrary value?) in place of the nulls. This doesn't cause problems working within our own ecosystem presumably because we maintain a mask that tells us where the nulls live and pass it around when constructing new Series object, e.g. this works:

`
>>> s = Series([1,2,3,None], dtype='object')
>>> s
0       1
1       2
2       3
3    null
dtype: int64
>>> s.data.mem.copy_to_host()
array([1, 2, 3, 0]) # becomes a zero
>>> s.astype(str).data.to_host()
['1', '2', '3', None]` # but we know it really isn't

This actually shields us against having problems casting around between datatypes, but causes issues when we call things like `to_pandas()` which seems to interpret the data more literally (which ends up causing problems with our testing suite which makes heavy use of this function):

`
>>> s.to_pandas()
0    1
1    2
2    3
3   -1
dtype: int64`

As such one could argue this is a problem with `to_pandas()` actually for both cases but at least in the object case we don't replace all the nulls with something else despite the presence of the mask to mark them.

I can look into fixing to_pandas in this PR if that seems like it would be helpful.

Reply:
We'd have to use nullable types in Pandas as opposed to the integral types which are very new and somewhat unsupported in general for now. I feel like returning those types will lead to a somewhat poor user experience so we shouldn't do that yet. In general I think the above behavior is fine, the only other way we could change it is to throw an exception if there's nulls as opposed to filling them with a sentinel value.